src/lib/Liquidity/LiquidityQueue.ts

```diff
@@ -882,15 +894,18 @@ export class LiquidityQueue {
         this._totalReserved.set(this.tokenId, newReserved);
     }
 
-    public distributeFee(_totalFee: u256): void {
-        // !!!!TODO: Add motoswap fee here
+    public distributeFee(totalFee: u256): void {
+        // TODO: Do we need to updateTotalReserve? What about virtualReserve?
         // if sending token to someone else, we need to call updateTotalReserve
         // this.virtualTokenReserve = SafeMath.add(this.virtualTokenReserve, totalFee);
+        // TODO: What is this split? Is it outdated?
         // If you want an 80/20 split:
         // const feeLP = SafeMath.div(SafeMath.mul(totalFee, u256.fromU64(80)), u256.fromU64(100));
         // const feeMoto = SafeMath.sub(totalFee, feeLP);
         // this.virtualTokenReserve = SafeMath.add(this.virtualTokenReserve, feeLP);
         // TransferHelper.safeTransfer(this.token, MOTOSWAP, feeMoto);
+
+        TransferHelper.safeTransfer(this.token, this.stakingContractAddress.value, totalFee);
```

Comment:
So, what's going on here? The totalFee is computed in a very long and convoluted way and I'm not sure if I should be sending 0.2% of this amount to the staking contract, or just send the whole thing.

Testing with a simple value of 10000 tokens, sending the whole fee results in the correct number, but I think that's because of the BaseBPs... I tried adding in the 20 BPs fee into the dynamic fee calculation (`DynamicFee.getDynamicFeeBP`) but then extracting it here becomes a hassle. It becomes messy to the point where I doubt it's the correct way of going ahead with it.
Please advise :)